src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,67 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
+
+import cn from 'classnames';
+import { Comment } from '../types/Comment';
+import { postComment } from '../utils/requestService';
+import { AppContext } from './Context';
 
 export const NewCommentForm: React.FC = () => {
+  const [name, setName] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [email, setEmail] = useState('');
+  const [emailError, setEmailError] = useState(false);
+  const [body, setBody] = useState('');
+  const [bodyError, setBodyError] = useState(false);
+
+  const appContext = useContext(AppContext);
+
+  const {
+    selectedPostId,
+    setNewComment,
+  } = appContext;
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!selectedPostId) {
+      return;
+    }
+
+    if (!name) {
+      setNameError(true);
+    }
+
+    if (!email) {
+      setEmailError(true);
+    }
+
+    if (!body) {
+      setBodyError(true);
+    }
+
+    if (!name || !email || !body) {
+      setBody('');
+
+      return;
+    }
+
+    const newComment: Comment = {
+      id: 0,
+      postId: selectedPostId,
+      name,
+      email,
+      body,
+    };
+
+    postComment(newComment)
```

Comment:
when adding a new comment the list of comments "jumps" - no need to show loader there, see how [the working example](https://mate-academy.github.io/react_dynamic-list-of-posts) works

[Screencast from 29-11-23 20:37:24.webm](https://github.com/mate-academy/react_dynamic-list-of-posts/assets/86923089/1874c738-957d-46f2-adea-15dcefea5e0c)

Also, you should clear the body field after that